dev/com.ibm.ws.webcontainer/src/com/ibm/ws/webcontainer/osgi/collaborator/CollaboratorHelperImpl.java

```diff
@@ -127,47 +126,95 @@ public IWebAppTransactionCollaborator getWebAppTransactionCollaborator() {
+          }
+
+          for (WebAppInvocationCollaborator inv : webAppInvCollabs)
+          {
+              inv.postInvoke(cmd,request,response);
+          }
+      }
+
+      if (com.ibm.ejs.ras.TraceComponent.isAnyTracingEnabled() && logger.isLoggable(Level.FINE)) {
+          logger.exiting(CLASS_NAME, "doInvocationCollaboratorsPostInvoke(webAppInv,cmd, req, resp)");
       }
-    }
   }
 
   public void doInvocationCollaboratorsPreInvoke(IInvocationCollaborator[] webAppInvocationCollaborators, com.ibm.wsspi.webcontainer.metadata.WebComponentMetaData cmd)
   {
-    if (webAppInvCollabs != null && !webAppInvCollabs.isEmpty())
-    {
-      for (WebAppInvocationCollaborator inv : webAppInvCollabs)
+      if (com.ibm.ejs.ras.TraceComponent.isAnyTracingEnabled() && logger.isLoggable(Level.FINE)) {
+          logger.entering(CLASS_NAME, "doInvocationCollaboratorsPreInvoke(webAppInv,cmd)");
+      }
+
+      if (webAppInvCollabs != null && !webAppInvCollabs.isEmpty())
       {
-        inv.preInvoke(cmd);
+          if (com.ibm.ejs.ras.TraceComponent.isAnyTracingEnabled() && logger.isLoggable(Level.FINE)) {
+              logger.logp(Level.FINE, "CollaboratorHelperImpl", "doInvocationCollaboratorsPreInvoke", " preInvoke WebAppInvocationCollaborator");
+          }
+
+          for (WebAppInvocationCollaborator inv : webAppInvCollabs)
+          {
+              inv.preInvoke(cmd);
+          }
+      }
+
+      if (com.ibm.ejs.ras.TraceComponent.isAnyTracingEnabled() && logger.isLoggable(Level.FINE)) {
+          logger.exiting(CLASS_NAME, "CollaboratorHelperImpl", "doInvocationCollaboratorsPreInvoke(webAppInv,cmd)");
       }
-    }
   }
 
   public void doInvocationCollaboratorsPostInvoke(IInvocationCollaborator[] webAppInvocationCollaborators, com.ibm.wsspi.webcontainer.metadata.WebComponentMetaData cmd)
   {
-    if (webAppInvCollabs != null && !webAppInvCollabs.isEmpty())
-    {
-      for (WebAppInvocationCollaborator inv : webAppInvCollabs)
+      if (com.ibm.ejs.ras.TraceComponent.isAnyTracingEnabled() && logger.isLoggable(Level.FINE)) {
+          logger.entering(CLASS_NAME, "CollaboratorHelperImpl", "doInvocationCollaboratorsPostInvoke(webAppInv,cmd)");
+      }
+
+      if (webAppInvCollabs != null && !webAppInvCollabs.isEmpty())
       {
-        inv.postInvoke(cmd);
+          if (com.ibm.ejs.ras.TraceComponent.isAnyTracingEnabled() && logger.isLoggable(Level.FINE)) {
+              logger.logp(Level.FINE, "CollaboratorHelperImpl", "doInvocationCollaboratorsPostInvoke", " postInvoke WebAppInvocationCollaborator");
+          }
+
+          for (WebAppInvocationCollaborator inv : webAppInvCollabs)
+          {
+              inv.postInvoke(cmd);
+          }
+      }
+
+      if (com.ibm.ejs.ras.TraceComponent.isAnyTracingEnabled() && logger.isLoggable(Level.FINE)) {
+          logger.exiting(CLASS_NAME, "CollaboratorHelperImpl", "doInvocationCollaboratorsPostInvoke(webAppInv,cmd)");
```

Comment:
existing takes class,method,result, so the second parameter should be the method name for all of the existing methods in this PR and you could just have `null` for the third parameter.

Reply:
Similar comment for the entering traces but the third parameter is for the parameters passed into the method.